tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,689 @@
+
+#### Exception 2: Changelog Auto-Generation (Data-Plane)
+
+**Description:** No automatic generation for data-plane; manual editing expected.
+**Impact:** Tool returns success with a `next_steps` message prompting manual edits.
+**Workaround:** Future LLM integration (post milestone 1).
+
+#### Language-Specific Limitations
+
+| Language   | Limitation | Impact | Workaround |
+|------------|------------|--------|------------|
+| JavaScript | Code formatting isn't hooked up to the emitter | Noisy diffs or linter failures | Run code formatting step in the `update-metadata.ps1` script |
+
+---
+
+## Design Proposal
+
+### Overview
+
+Implement four individual CLI + MCP tools. Each tool:
+
+1. Resolves package path & language.
+2. Loads the configuration for the language.
+3. Chooses inline command over script path if both present.
+4. Executes the relevant sub-script(s) in an ordered, fault-isolated manner.
+5. Aggregates structured result JSON with a `result`, human-readable `message`, and optional `next_steps` hint guiding subsequent tool invocation.
+
+Provides a singular CLI + MCP command that orchestrates the four tools (update-ci → update-changelog → update-version → update-metadata) and returns a combined JSON summary.
+
+### Detailed Design
+
+The four tools are intentionally small, composable units. Each emits a machine-readable JSON payload to enable deterministic chaining by an agent or CI workflow. All tools share common behaviors:
+
+| Area | Expected Behavior |
+|--------|-------------------|
+| Invocation Mode | CLI and MCP (agent) alias; same semantics |
+| Input Path Validation | Must exist and contain a recognizable SDK language structure; errors early if not |
+| Config Resolution | Load `swagger_to_sdk_config.json`; prefer `inline command` over `script path` when both are configured |
+| Timeout (default) | 5 minutes per tool invocation |
+| Output Schema | JSON with fields: result, message (see schema below) |
+| Idempotency | Multiple successive runs produce identical filesystem state except for timestamps/logs |
+| Script Implementation | MUST be PowerShell `.ps1`. SHOULD accept named parameters (e.g., `sdkRepoPath`). MUST use exit code `0` for success/partial/noop; non-zero only on outright failure. Legacy non-PS scripts should be wrapped by a small PowerShell shim. Scripts MUST write a concise stderr line prefixed with `ERROR:`; include suggested mitigation steps and links to docs in the message where possible. |
+| Script Execution Environment | All scripts are executed using PowerShell Core (`pwsh`) terminal |
+| Next Steps Hint | Plain language phrase embedded in `message` (e.g., `next_steps: update version`) |
+| Missing Required Tools | If a required external tool is not present (for example: `dotnet`, `gofmt`, `pwsh`), prompt the user to run the `verify-setup` tool which validates and documents missing prerequisites. |
+
+#### 1. Changelog Update Tool
+
+Name (CLI): `azsdk package update changelog`
```

Comment:
IMO I think this should be "update changelog-content" or something and all it does is update the content

Reply:
Agree, I thought about using it since the update only involves the content.

Reply:
Updated.